components/script/dom/htmlimageelement.rs

```diff
@@ -866,12 +921,10 @@ impl HTMLImageElement {
         let src = elem.get_url_attribute(&local_name!("src"));
         let base_url = document.base_url();
 
-        // https://html.spec.whatwg.org/multipage/#reacting-to-dom-mutations
-        // Always first set the current request to unavailable,
-        // ensuring img.complete is false.
         {
-            let mut current_request = self.current_request.borrow_mut();
-            current_request.state = State::Unavailable;
+            if let Some(ref mut pending_request) = *self.pending_request.borrow_mut() {
```

Comment:
I think we actually need the opposite logic:
```rust
let pending_request = self.pending_request.borrow_mut();
if pending_request.is_none() {
    *pending_request = Some(ImageRequest::new());
    pending_request.state = State::Unvailable;
}
```
This means we won't overwrite any already state that may already exist here, but if there is only a complete image then we'll add a new request that will trigger the conditional in HTMLImageElement::Complete correctly.

Reply:
Ok, that makes sense.  I just pushed a commit with the following:

```rust
let mut pending_opt = self.pending_request.borrow_mut();
if pending_opt.is_none() {
    *pending_opt = Some(ImageRequest::new());
}
```

WPT tests with back-to-back `<img>.complete` calls failed when I tried this locally on the first `<img>.complete` of each test, but perhaps CI WPT will prove that this change is fine.

This is different than your suggestion in that it doesn't explicitly set state to `Unavailable`, but `ImageRequest::new()` already sets the newly created image request state to `Unavailable`, so it should be effectively the same.  I tried to construct a new pending request and then explicitly set its state to `Unavailable`, but got [this error](https://stackoverflow.com/questions/58443400/rust-type-must-be-rooted).  If explicitly setting this state is desirable, I can continue my finaggling.